### PR TITLE
Activate hard forks in tests

### PR DESCRIPTION
**Pull request by karlb**
We should run tests with all hard forks enables by default. Otherwise, we will not test the behaviour that will eventually be used on mainnet. Another advantage is that we can undo changes compared to geth the we needed temporarily, but are unnecessary on when using the latest fork.

### CODE REVIEW

File: eth/tracers/api_test.go [hunk 1]
```diff
@@ -674,7 +675,7 @@ func TestCallTraceTransactionPrecompileTransfer(t *testing.T) {
 	gas := params.TxGas * 2
 	api := NewAPI(newTestBackend(t, 1, genesis, func(i int, b *core.BlockGen) {
 		// Transfer via transfer precompile by sending tx from GoldToken addr
-		tx, _ := types.SignTx(types.NewTransaction(uint64(i), transferPrecompile, big.NewInt(0), gas, big.NewInt(3), nil, nil, nil, data), signer, goldToken.key)
+		tx, _ := types.SignTx(types.NewTransaction(uint64(i), transferPrecompile, big.NewInt(0), gas, b.MinimumGasPrice(nil), nil, nil, nil, data), signer, goldToken.key)
```
Comment (gastonponti) [inline eth/tracers/api_test.go:678]: `b.MinimumGasPrice(3)` ?
Does that 3 changes something?

File: params/config.go [hunk 2]
```diff
@@ -580,3 +600,59 @@ func (c *ChainConfig) Rules(num *big.Int) Rules {
...
+		EIP155Block:         copyBigIntOrNil(c.EIP155Block),
+		EIP158Block:         copyBigIntOrNil(c.EIP158Block),
+		ByzantiumBlock:      copyBigIntOrNil(c.ByzantiumBlock),
+		ConstantinopleBlock: copyBigIntOrNil(c.ConstantinopleBlock),
+		PetersburgBlock:     copyBigIntOrNil(c.PetersburgBlock),
+		IstanbulBlock:       copyBigIntOrNil(c.IstanbulBlock),
+		ChurritoBlock:       copyBigIntOrNil(c.ChurritoBlock),
+		DonutBlock:          copyBigIntOrNil(c.DonutBlock),
+		EspressoBlock:       copyBigIntOrNil(c.EspressoBlock),
+
```
Comment (gastonponti) [inline params/config.go:636]: GFork?

File: core/genesis.go [hunk 3]
```diff
@@ -382,7 +382,7 @@ func DefaultAlfajoresGenesisBlock() *Genesis {
 // DeveloperGenesisBlock returns the 'geth --dev' genesis block.
 func DeveloperGenesisBlock(period uint64) *Genesis {
 	// Override the default period to the user requested one
-	config := *params.DeveloperChainConfig
+	config := *params.IstanbulTestChainConfig
```
Comment (gastonponti) [inline core/genesis.go:385]: Istanbul and not Test, right? (Just checking to be 100% sure)

**Comment by karlb** [comment]
All comments are marked as resolved. So is there an open todo here?

**Review by gastonponti — APPROVED** [review]
LGTM

**Comment by gastonponti** [comment]
> All comments are marked as resolved. So is there an open todo here?

No no. I'm merging it right now

**Pull request merged**